include/envoy/singleton/instance.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+namespace Envoy {
+namespace Singleton {
```

Comment:
Can you cleanup the TODO in https://github.com/lyft/envoy/blob/master/source/common/common/singleton.h#L19 and and add a pointer in the comments on that class to where to go for mutable singletons.

If we didn't want cleanup on last dropped reference, and only had default constructors, we could do something simple like what is done for `ConstSingleton` perhaps.